Backend/src/main/java/com/SYSC4806_Group13/SYSC4806_Project/Controllers/ListingRestController.java

```diff
@@ -0,0 +1,215 @@
+        }
+    }
+
+    @PostMapping("/listings")
+    public Listing addListing(@RequestBody Map<String, Object> payload) {
+
+        Integer sellerUserId = (Integer) payload.get("sellerUserId");
+        if (sellerUserId == null) {
+            throw new MissingAttributeException("Request body must contain 'sellerUserId'");
+        }
+
+        String isbn = (String) payload.get("isbn");
+        if (isbn == null) {
+            throw new MissingAttributeException("Request body must contain 'isbn'");
+        }
+
+        String title = (String) payload.get("title");
+        if (title == null) {
+            throw new MissingAttributeException("Request body must contain 'title'");
+        }
+
+        String price = (String) payload.get("price");
+        Float floatPrice;
+
+        if (price == null) {
+            throw new MissingAttributeException("Request body must contain 'price'");
+        }
+
+        try {
+            floatPrice = Float.valueOf(price);
+        } catch (NumberFormatException e) {
+            throw new MissingAttributeException("Attribute 'price' must be formatted in decimal as a string (ex. '1.5')");
+        }
+
+        String author = (String) payload.get("author");
+        if (author == null) {
+            throw new MissingAttributeException("Request body must contain 'author'");
+        }
+
+        String publisher = (String) payload.get("publisher");
+        if (publisher == null) {
+            throw new MissingAttributeException("Request body must contain 'publisher'");
+        }
+
+        String description = (String) payload.get("description");
+        if (description == null) {
+            throw new MissingAttributeException("Request body must contain 'description'");
+        }
+
+        Integer inventory = (Integer) payload.get("inventory");
+        if (inventory == null) {
+            throw new MissingAttributeException("Request body must contain 'inventory'");
+        }
+
+        String releaseDate = (String) payload.get("releaseDate");
+        if (releaseDate == null) {
+            throw new MissingAttributeException("Request body must contain 'releaseDate'");
+        }
+
+        String coverImage = (String) payload.get("coverImage");
+        if (coverImage == null) {
+            throw new MissingAttributeException("Request body must contain 'coverImage'");
+        }
+
+        Boolean isActive = true;
+
+        Listing listing = new Listing(
+                sellerUserId.longValue(),
+                isbn,
+                title,
+                floatPrice,
+                author,
+                publisher,
+                description,
+                inventory,
+                releaseDate,
+                isActive,
+                coverImage
+        );
+
+        repo.save(listing);
+
+        return listing;
+    }
+
+    @PutMapping("/listings")
+    public Map<String, Object> putListing(@RequestBody Map<String, Object> payload) {
```

Comment:
All string params should be checked for non-emptyness and they should also be checked to make sure there are no pure-whitespace strings. Same goes with PUT

Reply:
Added blankness checks for all string attributes for POST and PUT